Test/Algorithms/DirectSearch/CMA.cpp

```diff
@@ -14,6 +14,51 @@
 
 using namespace shark;
 
+
+struct MultiplicativeNoisySphere : public SingleObjectiveFunction {
+	
+	MultiplicativeNoisySphere(std::size_t numberOfVariables, double sigma):m_numberOfVariables(numberOfVariables), m_sigma(sigma) {
+		m_features |= CAN_PROPOSE_STARTING_POINT;
+		m_features |= IS_NOISY;
+	}
+
+	/// \brief From INameable: return the class name.
+	std::string name() const
+	{ return "MultiplicativeNoisySphere"; }
+
+	std::size_t numberOfVariables()const{
+		return m_numberOfVariables;
+	}
+	
+	bool hasScalableDimensionality()const{
+		return true;
+	}
+
+	void setNumberOfVariables( std::size_t numberOfVariables ){
+		m_numberOfVariables = numberOfVariables;
+	}
+
+	SearchPointType proposeStartingPoint() const {
+		RealVector x(numberOfVariables());
+
+		for (std::size_t i = 0; i < x.size(); i++) {
+			x(i) = Rng::gauss(0,1);
+		}
+		return x;
+	}
+
+	double eval(SearchPointType const& x) const {
+		SIZE_CHECK(x.size() == numberOfVariables());
+		m_evaluationCounter++;
+		double f = norm_sqr(x);
+		double noise = exp(Rng::gauss(0,m_sigma));
```

Comment:
I like the log-normal approach, but the standard thing in analysis papers is something like
f *= (1 + sigma * gauss)
or
f *= (1 + sigma * uniform)
That's actually very different since noisy values can become negative in some cases, which completely destroys elitist methods (well, usually step size adaptation kills the method already at much smaller noise levels).